### PR TITLE
Added beginner 2D movement tutorial

### PR DESCRIPTION
**Pull request by cbscribe**
After seeing the same questions pop up again and again in Q&A and Discord, I decided to make a brief tutorial demonstrating four basic types of 2D movement.

### CODE REVIEW

**Comment by skyace65** [comment]
Whenever this gets merged I'll get to work on C# code for the tutorial

File: tutorials/2d/2d_movement.rst [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+2D Movement Overview
+====================
+
+Introduction
+------------
+
+Every beginner has been there: "How do I move my character?" Depending on the
+style of game you're making, you may have special requirements, but in general
+the movement in most 2D games is based on a small number of
```
Comment (NathanLovato) [inline tutorials/2d/2d_movement.rst]: Missing end of the sentence

File: tutorials/2d/2d_movement.rst [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+We'll use :ref:`KinematicBody2D <class_KinematicBody2D>` for these examples,
+but the principles will apply to other node types (Area2D, RigidBody2D) as well.
+
+Setup
+-----
+
+Each example below uses the same scene setup. Start with a ``KinematicBody2D`` with two
+children: ``Sprite`` and ``CollisionShape2D``. You can use the Godot icon ("icon.png")
+for the Sprite's texture or use any other 2D image you have available.
```
Comment (NathanLovato) [inline tutorials/2d/2d_movement.rst]: Have available is redundant

File: tutorials/2d/2d_movement.rst [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+input actions (see :ref:`InputEvent <doc_inputevent>` for details):
+
+.. image:: img/movement_inputs.png
+
+8-Way Movement
+--------------
+
+In this scenario, you want the user to press the four directional keys (up/left/down/right
+or W/A/S/D) and move in the selected direction. The name "8-way movement" comes from the
+fact that diagonal movement can be chosen by pressing two keys at once.
```
Comment (NathanLovato) [inline tutorials/2d/2d_movement.rst]: A little odd to me. The player can move diagonally by pressing 2 arrow keys simultaneously/at once.

File: tutorials/2d/2d_movement.rst [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+        if Input.is_action_pressed('left'):
+            rot_dir -= 1
+        if Input.is_action_pressed('down'):
+            velocity = Vector2(-speed, 0).rotated(rotation)
+        if Input.is_action_pressed('up'):
+            velocity = Vector2(speed, 0).rotated(rotation)
+
+    func _physics_process(delta):
+        get_input()
+        rotation += rot_dir * rot_speed * delta
```
Comment (NathanLovato) [inline tutorials/2d/2d_movement.rst]: Personal taste here but for educational purposes I'd rather use plain English variable names: rotation_direction etc. More as a rule of thumb for clarity, esp for beginners

File: tutorials/2d/2d_movement.rst [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+        if Input.is_action_pressed('down'):
+            velocity = Vector2(-speed, 0).rotated(rotation)
+        if Input.is_action_pressed('up'):
+            velocity = Vector2(speed, 0).rotated(rotation)
+
+    func _physics_process(delta):
+        get_input()
+        move_and_slide(velocity)
+
+Note that the :ref:`Node2D <class_Node2D>` ``look_at()`` method exists just
```
Comment (NathanLovato) [inline tutorials/2d/2d_movement.rst]: Sounds a little convoluted. We can use look_at to make the character/ship rotate towards the mouse

File: tutorials/2d/2d_movement.rst (hunk 1 above)
Comment (cbscribe) [reply tutorials/2d/2d_movement.rst]: Oops I accidentally a word.

File: tutorials/2d/2d_movement.rst (hunk 3 above)
Comment (cbscribe) [reply tutorials/2d/2d_movement.rst]: Reworded.

File: tutorials/2d/2d_movement.rst (hunk 4 above)
Comment (cbscribe) [reply tutorials/2d/2d_movement.rst]: Yeah, I agree. I took this code from an older project and should have renamed them.

File: tutorials/2d/2d_movement.rst (hunk 5 above)
Comment (cbscribe) [reply tutorials/2d/2d_movement.rst]: Reworded

**Pull request merged**